mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageDAO.java

```diff
@@ -182,8 +184,8 @@ private PreparedStatement prepareDelete(Session session) {
             byte[] headerContent = IOUtils.toByteArray(message.getHeaderContent());
             byte[] bodyContent = IOUtils.toByteArray(message.getBodyContent());
 
-            Mono<BlobId> bodyFuture = blobStore.save(blobStore.getDefaultBucketName(), bodyContent);
-            Mono<BlobId> headerFuture = blobStore.save(blobStore.getDefaultBucketName(), headerContent);
+            Mono<BlobId> bodyFuture = blobStore.save(blobStore.getDefaultBucketName(), bodyContent, LowCost);
```

Comment:
why not SizeBased?

Reply:
I hesitated on this but I believe this was an error in the original ADR (that should be modified as well)...

 - Headers are expected to be small, often read
 - Body 'can' be small and **are rarely read**

I'm convinced performance/cost ratio to be worth it for headers, I am way less convinced about it regarding bodies.

Reply:
I was going to say "let measure read rate in production" then think "let decide at runtime with statistics".
In any case, that would be good to be able to configure it and have a migration process
All that make me think it's a good idea but maybe a premature one.

Reply:
> All that make me think it's a good idea but maybe a premature one.

This sentence is ambiguate. Can you define what is in the scope of ` it's a good idea` ?

Reply:
> and have a migration process

That's not necessary:
 - new policies are applied to newly saved blobs
 - read end up trying both blobStores hence no migration process is required for correctness.

This looks acceptable to me.

Let's add this as configuration options.